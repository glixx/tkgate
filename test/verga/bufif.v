module top;
  integer i,j;
  reg a,c;
  wire w,x,y,z;

  function getbit;
  input [31:0] i;

    case (i)
      0: getbit = 1'b0;
      1: getbit = 1'b1;
      2: getbit = 1'bx;
      3: getbit = 1'bz;
      4: getbit = 1'bl;
      5: getbit = 1'bh;
    endcase
  endfunction

  initial
    $monitor("%4d: a=%b c=%b : pmos=%b nmos=%b bufif1=%b bufif0=%b",$time,a,c,w,x,y,z);

  initial
    for (i = 0;i < 6;i = i + 1)
      begin
	a = getbit(i);
	for (j = 0;j < 6;j = j + 1)
	  begin
	    c = getbit(j);
	    #10;
	  end
      end
 
  pmos    b(w,a,c);
  nmos    b(x,a,c);
  bufif1  b(y,a,c);
  bufif0  b(z,a,c);

endmodule
