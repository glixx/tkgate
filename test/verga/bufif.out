   0: a=0 c=0 : pmos=0 nmos=z bufif1=z bufif0=0
  10: a=0 c=1 : pmos=z nmos=0 bufif1=0 bufif0=z
  20: a=0 c=x : pmos=L nmos=L bufif1=L bufif0=L
  30: a=0 c=z : pmos=L nmos=L bufif1=L bufif0=L
  40: a=0 c=L : pmos=L nmos=L bufif1=L bufif0=L
  50: a=0 c=H : pmos=L nmos=L bufif1=L bufif0=L
  60: a=1 c=0 : pmos=1 nmos=z bufif1=z bufif0=1
  70: a=1 c=1 : pmos=z nmos=1 bufif1=1 bufif0=z
  80: a=1 c=x : pmos=H nmos=H bufif1=H bufif0=H
  90: a=1 c=z : pmos=H nmos=H bufif1=H bufif0=H
 100: a=1 c=L : pmos=H nmos=H bufif1=H bufif0=H
 110: a=1 c=H : pmos=H nmos=H bufif1=H bufif0=H
 120: a=x c=0 : pmos=x nmos=z bufif1=z bufif0=x
 130: a=x c=1 : pmos=z nmos=x bufif1=x bufif0=z
 140: a=x c=x : pmos=x nmos=x bufif1=x bufif0=x
 150: a=x c=z : pmos=x nmos=x bufif1=x bufif0=x
 160: a=x c=L : pmos=x nmos=x bufif1=x bufif0=x
 170: a=x c=H : pmos=x nmos=x bufif1=x bufif0=x
 180: a=z c=0 : pmos=z nmos=z bufif1=z bufif0=x
 190: a=z c=1 : pmos=z nmos=z bufif1=x bufif0=z
 200: a=z c=x : pmos=z nmos=z bufif1=x bufif0=x
 210: a=z c=z : pmos=z nmos=z bufif1=x bufif0=x
 220: a=z c=L : pmos=z nmos=z bufif1=x bufif0=x
 230: a=z c=H : pmos=z nmos=z bufif1=x bufif0=x
 240: a=L c=0 : pmos=L nmos=z bufif1=z bufif0=x
 250: a=L c=1 : pmos=z nmos=L bufif1=x bufif0=z
 260: a=L c=x : pmos=L nmos=L bufif1=x bufif0=x
 270: a=L c=z : pmos=L nmos=L bufif1=x bufif0=x
 280: a=L c=L : pmos=L nmos=L bufif1=x bufif0=x
 290: a=L c=H : pmos=L nmos=L bufif1=x bufif0=x
 300: a=H c=0 : pmos=H nmos=z bufif1=z bufif0=x
 310: a=H c=1 : pmos=z nmos=H bufif1=x bufif0=z
 320: a=H c=x : pmos=H nmos=H bufif1=x bufif0=x
 330: a=H c=z : pmos=H nmos=H bufif1=x bufif0=x
 340: a=H c=L : pmos=H nmos=H bufif1=x bufif0=x
 350: a=H c=H : pmos=H nmos=H bufif1=x bufif0=x
